js/stdlib/ts/ALGO.ts

```diff
@@ -641,21 +651,26 @@ export class EventCache {
         .txType('appl')
 
     if (minRound) {
-      query = query.minRound(minRound + 1);
-    }
-    if (maxRound) {
-      query = query.maxRound(maxRound);
+      // If cache has: [100, 200]
+      // & querying  : [150, 1000]
+      // We already searched cache for [150, 200] so query network for [201, 1000]
+      query = query.minRound(Math.max(this.currentRound, minRound) + 1);
```

Comment:
Based on the next comment, I think that this could should not use `minRound`, but should use `filterRound`

Reply:
You really want `Math.max(this.currentRound + 1, minRound)` because the user really wants the lower bound to actually be `minRound` (i.e. `filterRound`)